_impls/thrussh.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Thrussh
+homepage: http://pijul.org/thrussh
+source-repository: https://nest.pijul.com/pijul_org/thrussh
+license: "[Apache-2.0](https://www.apache.org/licenses/LICENSE-2.0)"
+first-release:
+    date: 2016-07-01
+latest-release:
+    version: 0.10.3
+    date: 2017-05-07
+client: yes
+server: yes
+
+protocols:
+    cipher:
+        - chacha20-poly1305@openssh.com
+    compression:
+        - none
+    hostkey:
+        - rsa-sha2-256
+        - rsa-sha2-512
+        - ssh-ed25519                       # since 6.5
+    kex:
+        - curve25519-sha256@libssh.org      # since 6.5
+        - ecdh-sha2-nistp256                # since 5.7
+    mac:
```

Comment:
Thrussh does comply with the RFCs, see for instance https://tools.ietf.org/html/rfc4251#section-9.3.2

In particular:
> This protocol does allow the Data Integrity mechanism to be disabled.
   Implementers SHOULD be wary of exposing this feature for any purpose
   other than debugging.  Users and administrators SHOULD be explicitly
   warned anytime the "none" MAC is enabled.

This might be the case in TinySSH, but not in Thrussh: since the MAC is included in the cipher, it is never disabled.

It really isn't a bug to include MACs in ciphers. Combining these is hard to get right, especially when one wants to support multiple algorithms at each layer.